pkg/operation/care/webhook_remediation.go

```diff
@@ -181,28 +174,34 @@ func (r *WebhookRemediation) Remediate(ctx context.Context) error {
 	return flow.Parallel(fns...)(ctx)
 }
 
-func mustRemediateTimeoutSeconds(timeoutSeconds *int32) bool {
-	return timeoutSeconds == nil || *timeoutSeconds > WebhookMaximumTimeoutSecondsNotProblematic
-}
-
-func mustRemediateSelector(
-	failurePolicy *admissionregistrationv1.FailurePolicyType,
+func getMatchingRules(
 	rules []admissionregistrationv1.RuleWithOperations,
 	objectSelector, namespaceSelector *metav1.LabelSelector,
-) (
-	bool,
-	matchers.WebhookConstraintMatcher,
-) {
-	if failurePolicy == nil || *failurePolicy != admissionregistrationv1.Ignore {
-		for _, rule := range rules {
-			for _, matcher := range matchers.WebhookConstraintMatchers {
-				if matcher.Match(rule, objectSelector, namespaceSelector) {
-					return true, matcher
-				}
+) []webhookmatchers.WebhookConstraintMatcher {
+	var matchers []webhookmatchers.WebhookConstraintMatcher
+	for _, rule := range rules {
+		for _, matcher := range webhookmatchers.WebhookConstraintMatchers {
+			if matcher.Match(rule, objectSelector, namespaceSelector) {
+				matchers = append(matchers, matcher)
 			}
 		}
 	}
-	return false, matchers.WebhookConstraintMatcher{}
+	return matchers
+}
+
+func mustRemediateTimeoutSeconds(timeoutSeconds *int32) bool {
+	return timeoutSeconds == nil || *timeoutSeconds > WebhookMaximumTimeoutSecondsNotProblematic
+}
+
+func mustRemediateSelectors(matchers []webhookmatchers.WebhookConstraintMatcher) bool {
+	return len(matchers) > 0
+}
+
+func mustRemediateFailurePolicy(matchers []webhookmatchers.WebhookConstraintMatcher) bool {
+	for _, matcher := range matchers {
+		return matcher.NamespaceLabels == nil && matcher.ObjectLabels == nil
+	}
+	return false
 }
```

Comment:
@ary1992 and me discovered an issue in `mustRemediateFailurePolicy`. The current implementation would only return true if the first matcher of the array satisfies the condition `matcher.NamespaceLabels == nil && matcher.ObjectLabels == nil` 

```suggestion
func mustRemediateFailurePolicy(matchers []webhookmatchers.WebhookConstraintMatcher) bool {
	for _, matcher := range matchers {
		if matcher.NamespaceLabels == nil && matcher.ObjectLabels == nil {
			return true
		}
	}
	return false
}
```

Reply:
Ooops :) nice catch!